### ISSUE TITLE
[FP] Foreach source to iterate over on stdClass

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1287]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         |  Php Inspections (EA Ultimate), 2.0.14 |

Current behaviour
--
For the following code:
```php
function scope(){
    $a = new \stdClass();
    $a->a = 'a';
    $a->b = 'b';
    foreach($a as $param){
        echo $param;
    }
}
```

The plugin launch a warning "Iterates over '\stdClass' properties (probably should implement one of Iterator interfaces).".

However, there is no way to make stdClass implement an interface. This is problematic for example with classes like SoapClient who will output stdClass to structure the return. The foreach loop is very useful in those cases.

Thanks

**Comment by funivan** [comment]
You can wrap `foreach` over stdClass in one function and suppress inspection in 1 place.
But maybe @kalessil can make a list of ignored classes (configured by user)

**Comment by orklah** [comment]
> You can wrap `foreach` over stdClass in one function and suppress inspection in 1 place.
> But maybe @kalessil can make a list of ignored classes (configured by user)

If I understand, you say you want to keep this warning for everyone by default?

I can understand why you would want this warning on a random class : if it doesn't implement Iterator, every attribute will show up in the loop. That means if the developper is not careful, a simple attribute creation could break code elsewhere. However, on stdClass, the whole point of the loop is to go over every attribute.

Maybe there is a case I don't see, but for me, every foreach on a stdClass do exactly what it should do. There was even a proposal on PHP internals to make stdClass implements Traversable by default:
https://externals.io/message/103718

**Comment by kalessil** [comment]
Fixed!
